### PR TITLE
[SYCL][E2E] Add a test showing pre-existing swizzle-related bugs

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

File: sycl/test/basic_tests/vectors/swizzle.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// RUN: %clangxx -fsycl %s -o %t_default.out
+// RUN: %t_default.out
```
Comment (bader) [inline sycl/test/basic_tests/vectors/swizzle.cpp:2]: Shouldn't SYCL program executed be tested by e2e tests?

Comment (aelovikov-intel) [reply sycl/test/basic_tests/vectors/swizzle.cpp:2]: This is device agnostic and can run on the build system.

File: sycl/test/basic_tests/vectors/swizzle.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    // expected-error-re@+1 {{no template named 'swizzle' in {{.*}}}}
+    auto tmp = sw.swizzle<1, 0>();
+    assert(tmp[0] == 3);
+    assert(tmp[1] == 2);
+  }
+
+  {
+    // expected-error-re@+1 {{no template named 'swizzle' in {{.*}}}}
+    auto tmp = (sw + sw).swizzle<1, 0>();
+    std::cout << tmp[0] << " " << tmp[1] << std::endl;
```
Comment (uditagarwal97) [inline sycl/test/basic_tests/vectors/swizzle.cpp:35]: `std::cout` should be removed?

**Review by uditagarwal97 — APPROVED** [review]
LGTM.

File: sycl/test/basic_tests/vectors/swizzle.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (aelovikov-intel) [inline sycl/test/basic_tests/vectors/swizzle.cpp:36]:
```suggestion
    auto tmp = (sw + sw).swizzle<1, 0>();

```

File: sycl/test/basic_tests/vectors/swizzle.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    auto tmp = sw.swizzle<1, 0>();
+    assert(tmp[0] == 3);
+    assert(tmp[1] == 2);
+  }
+
+  {
+    // expected-error-re@+1 {{no template named 'swizzle' in {{.*}}}}
+    auto tmp = (sw + sw).swizzle<1, 0>();
+
+    assert(tmp[0] == 6);
```
Comment (aelovikov-intel) [inline sycl/test/basic_tests/vectors/swizzle.cpp:36]:
```suggestion
    auto tmp = (sw + sw).swizzle<1, 0>();
    assert(tmp[0] == 6);
```

**Pull request merged**